server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3938,7 +3943,12 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
                     }
                     // update password for this host
                     final DetailVO nv = _detailsDao.findDetail(h.getId(), ApiConstants.USERNAME);
-                    if (nv.getValue().equals(command.getUsername())) {
+                    if (nv == null) {
+                        final DetailVO nvu = new DetailVO(h.getId(), ApiConstants.USERNAME, command.getUsername());
```

Comment:
@ravening should we use 'userNameWithoutSpaces' instead of cmd.getUserName() ?

Reply:
yes correct. probably i missed it. will change that